pkg/controller/autoimport/manager.go

```diff
@@ -46,20 +53,37 @@ func add(importSecretInformer, autoImportSecretInformer cache.SharedIndexInforme
 		return err
 	}
 
-	// watch the import secrets
+	// watch the klusterlet manifest works
 	if err := c.Watch(
-		source.NewImportSecretSource(importSecretInformer),
-		&source.ManagedClusterSecretEventHandler{},
+		&runtimesource.Kind{Type: &workv1.ManifestWork{}},
```

Comment:
I think you should watch import secret and work both. The key could be namespace.

Reply:
the key is namespace (`managedClusterName := request.Namespace`) , so we don't need a special handling

ok, add the import secret back, it will help to resolve corner case